apps/papra-server/src/modules/tagging-rules/tagging-rules.routes.ts

```diff
@@ -175,3 +176,38 @@ function setupUpdateTaggingRuleRoute({ app, db }: RouteDefinitionContext) {
     },
   );
 }
+
+function setupApplyTaggingRuleRoute({ app, db, taskServices }: RouteDefinitionContext) {
+  app.post(
+    '/api/organizations/:organizationId/tagging-rules/:taggingRuleId/apply',
+    requireAuthentication(),
+    validateParams(z.object({
+      organizationId: organizationIdSchema,
+      taggingRuleId: taggingRuleIdSchema,
+    })),
+    async (context) => {
+      const { userId } = getUser({ context });
+
+      const { organizationId, taggingRuleId } = context.req.valid('param');
+
+      const taggingRulesRepository = createTaggingRulesRepository({ db });
+      const organizationsRepository = createOrganizationsRepository({ db });
+
+      await ensureUserIsInOrganization({ userId, organizationId, organizationsRepository });
+
+      // Verify the tagging rule exists before enqueuing the task
+      await taggingRulesRepository.getOrganizationTaggingRule({ organizationId, taggingRuleId });
```

Comment:
This method does not throw on missing rule, add a throw on nil


```suggestion
      const { taggingRule } = await taggingRulesRepository.getOrganizationTaggingRule({ organizationId, taggingRuleId });
      
      if(isNil(taggingRule)) {
        throw ...
      }

```